borsh/src/de/mod.rs

```diff
@@ -294,6 +297,53 @@ impl BorshDeserialize for String {
     }
 }
 
+#[cfg(feature = "bigdecimal")]
+impl BorshDeserialize for bigdecimal::BigDecimal {
+    #[inline]
+    fn deserialize(buf: &mut &[u8]) -> Result<Self> {
+        let digits = bigdecimal::num_bigint::BigInt::deserialize(buf)?;
+        let scale = i64::deserialize(buf)?;
+        Ok(bigdecimal::BigDecimal::new(digits, scale))
+    }
+}
+
+#[cfg(feature = "bigdecimal")]
+impl BorshDeserialize for bigdecimal::num_bigint::BigInt {
+    #[inline]
+    fn deserialize(buf: &mut &[u8]) -> Result<Self> {
+        let sign = bigdecimal::num_bigint::Sign::deserialize(buf)?;
+        let digits = <Vec<u32>>::deserialize(buf)?;
+        Ok(bigdecimal::num_bigint::BigInt::new(sign, digits))
```

Comment:
checked the impl, there isn't any validation that this initialization or `from_bytes_le`. Not sure if just checking the last byte is sufficient for validating if the data is valid or not, but I can't think of a reason why it can't be that simple

Reply:
The condition we want here (with le bytes) is I think `if sign == NoSign { bytes.is_empty() } else { !bytes.is_empty() && bytes.last().unwrap() > 0}`

when we go to full decimals, we also need to check the extra degree of freedom from the exponent. I *though* that was equivalent to checking the first byte for zero, but that doesn’t work, as exponent is 10, not 256. Probably makes sense to study how normalize work and do the same logic.

Reply:
@austinabell @matklad are you sure something should be added to the current implementation? Please have a look at the new version

Reply:
Yes, I believe so. `from_bytes_le` does not reject if the bytes are not normalized, so we need this check to make sure there are no padded bytes/digits at the end of this

Reply:
Removed the padding